From fd16250083adda380c980e9c4304bec63c10de9d Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 17 Jul 2018 13:30:01 +0300
Subject: [PATCH 0466/1051] dmaengine: mv_xor_v2: convert callback to helper
 function

commit 5a80aff92ad28c5e3045b542576c1d08260606db upstream

This is in preparation of moving to a callback that provides results to the
callback for the transaction. The conversion will maintain current behavior
and the driver must convert to new callback mechanism at a later time in
order to receive results.

Signed-off-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Vinod Koul <vkoul@kernel.org>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/dma/mv_xor_v2.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index e7184985b2a2..14e2a7a2e80b 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -589,9 +589,8 @@ static void mv_xor_v2_tasklet(unsigned long data)
 			 */
 			dma_cookie_complete(&next_pending_sw_desc->async_tx);
 
-			if (next_pending_sw_desc->async_tx.callback)
-				next_pending_sw_desc->async_tx.callback(
-				next_pending_sw_desc->async_tx.callback_param);
+			dmaengine_desc_get_callback_invoke(
+					&next_pending_sw_desc->async_tx, NULL);
 
 			dma_descriptor_unmap(&next_pending_sw_desc->async_tx);
 		}
-- 
2.17.1

